src/main/java/org/spongepowered/api/service/permission/Subject.java

```diff
@@ -101,14 +114,17 @@
     SubjectData getTransientSubjectData();
 
     /**
-     * Test whether the subject is permitted to perform an action given as the
-     * given permission string.
+     * Test whether the subject is permitted to perform an action corresponding to the
+     * given permission string. This must return the same boolean equivalent as
+     * {@link #getPermissionValue(Set, String)}.
      *
      * @param contexts The set of contexts that represents the subject's current environment
      * @param permission The permission string
      * @return True if permission is granted
      */
-    boolean hasPermission(Set<Context> contexts, String permission);
+    default boolean hasPermission(Set<Context> contexts, String permission) {
```

Comment:
This should also mention the "format" permission strings are supposed to follow. 
(See `PermissionDescription` for more details)
And it should mention that there are no wildcard.* permissions.

The same applies for `getPermissionValue`